src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -75,7 +76,62 @@ internal static int CompareStringIgnoreCaseNonAscii(ref char strA, int lengthA,
             return OrdinalCasing.CompareStringIgnoreCase(ref strA, lengthA, ref strB, lengthB);
         }
 
+        private static bool EqualsIgnoreCase_Vector128(ref char charA, ref char charB, int length)
+        {
+            Debug.Assert(length >= Vector128<ushort>.Count);
+            Debug.Assert(Vector128.IsHardwareAccelerated);
+
+            nuint lengthU = (nuint)length;
+            nuint lengthToExamine = lengthU - (nuint)Vector128<ushort>.Count;
+            nuint i = 0;
+            Vector128<ushort> vec1;
+            Vector128<ushort> vec2;
+            do
+            {
+                vec1 = Vector128.LoadUnsafe(ref Unsafe.As<char, ushort>(ref charA), i);
+                vec2 = Vector128.LoadUnsafe(ref Unsafe.As<char, ushort>(ref charB), i);
+
+                if (!Utf16Utility.AllCharsInVector128AreAscii(vec1 | vec2))
+                {
+                    goto NON_ASCII;
+                }
+
+                if (!Utf16Utility.Vector128OrdinalIgnoreCaseAscii(vec1, vec2))
+                {
+                    return false;
+                }
+
+                i += (nuint)Vector128<ushort>.Count;
+            } while (i <= lengthToExamine);
+
+            // Use scalar path for trailing elements
+            return i == lengthU || EqualsIgnoreCase(ref Unsafe.Add(ref charA, i), ref Unsafe.Add(ref charB, i), (int)(lengthU - i));
+
+        NON_ASCII:
+            if (Utf16Utility.AllCharsInVector128AreAscii(vec1) || Utf16Utility.AllCharsInVector128AreAscii(vec2))
```

Comment:
> some examples of where ASCII chars can equal non-ASCII chars under an OrdinalIgnoreCase comparer

This comes up empty:
```C#
for (int ascii = 0; ascii < 128; ascii++)
    for (int nonAscii = 128; nonAscii <= char.MaxValue; nonAscii++)
        if (StringComparer.OrdinalIgnoreCase.Equals(((char)ascii).ToString(), ((char)nonAscii).ToString()))
            Console.WriteLine($"{ascii} === {nonAscii}");
```

Reply:
> @tarekgh Do you have thoughts on this? I thought there were some examples of where ASCII chars can equal non-ASCII chars under an OrdinalIgnoreCase comparer. (Or maybe we special cased all such characters away?)

Maybe Turkey's `i` ? Not sure it's related to OrdinalIgnoreCase though. Anyway, we already maintain this invariant that if an input is ASCII while the other is not then they're not equal, otherwise we wouldn't be able to unroll String.Equals/StartsWith for OrdinalIgnoreCase for const ASCII input. I recall that you also had doubts about that but can we please have performance when Ordinal* is specified 🙂

Reply:
I am not aware that with the ordinal comparisons there exist any equality of Ascii character with the non-Ascii characters. When creating the casing table, we filter out the Turkish `I`s and small letter `s` cases. 
https://github.com/dotnet/runtime/blob/a4e98d40a098a0a58a88c1205749050cfb5f9881/src/libraries/System.Private.CoreLib/Tools/GenUnicodeProp/CategoryCasingInfo.cs#L46